mesheryctl/internal/cli/root/system/logs.go

```diff
@@ -212,27 +244,45 @@ mesheryctl system logs meshery-istio
 					}
 					defer logs.Close()
 
-					buf := new(bytes.Buffer)
-					_, err = io.Copy(buf, logs)
-					if err != nil {
-						return fmt.Errorf("error in copy information from logs to buf")
-					}
-
-					// Append this to data to be printed
-					for _, str := range strings.Split(buf.String(), "\n") {
-						data = append(data, fmt.Sprintf("%s\t|\t%s", name, str))
+					if follow {
+						go func() {
+							for {
+								buf := make([]byte, BYTE_SIZE)
+								numBytes, err := logs.Read(buf)
+								if numBytes == 0 {
+									continue
+								}
+								if err == io.EOF {
+									break
+								}
+								if err != nil {
+									break
+								}
+								logChan <- logChanStruct{PodName: name, Message: buf[0:numBytes]}
+							}
+						}()
+					} else {
+						buf := new(bytes.Buffer)
+						_, err = io.Copy(buf, logs)
+						if err != nil {
+							return fmt.Errorf("error in copy information from logs to buf")
```

Comment:
This error message is leaky. You are talking about internal variables in user facing log messages. Replace them with appropriate messages that signal to the user that logs couldn't be fetched with possible reasons.

Reply:
You can also avoid creating another buffer variable by directly calling io.ReadAll on the logs in this case.